src/abyssal_tome/app.py

```
@@ -0,0 +1,772 @@
+    with Path("assets/processed_data.json").open(encoding="utf-8") as file:
+        data = json.load(file)
+    logging.info("JSON data loaded successfully.")
+    return data
+
+async def highlight_text(span: ft.TextSpan, search_term: str) -> list[ft.TextSpan]:
+    """
+    Highlight occurrences of a search term within a TextSpan, returning new spans with background color applied to matches.
+    
+    If the span represents a tag icon matching the search term, highlights the entire icon. Otherwise, splits the span into segments, applying a highlight style to each substring that matches the search term. Returns the resulting list of TextSpans, or the original span if no matches are found.
+    
+    Parameters:
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)